File: Q5_1282_8.py
for _ in range(int(input())):
  n,q=map(int,input().split())
  g=[[0]*1001 for _ in range(1001)]
  for _ in range(n):
    h,w=map(int,input().split())
    g[h][w]+=w*h
  for i in range(1001):
    for j in range(1001):
      if i > 0: g[i][j]+=g[i-1][j]
      if j > 0: g[i][j]+=g[i][j-1]
      if i>0 and j > 0: g[i][j]-=g[i-1][j-1]
  for _ in range(q):
    a,b,c,d=map(int,input().split())
    c-=1
    d-=1
    print(g[c][d]-g[c][b]-g[a][d]+g[a][b])

File: Q5_1282_8.cpp
#include <iostream>
#include <vector>
using namespace std; //This is Q9

int main() {
    int t;
    cin >> t; // Number of test cases

    while (t--) {
        int n;
        long long f, a, b;
        cin >> n >> f >> a >> b; // Input for n, f, a, b

        vector<long long> m(n + 1); // Moments array with an extra 0 at the start
        m[0] = 0; // First moment is 0

        for (int i = 1; i <= n; i++) {
            cin >> m[i]; // Input moments
        }

        // Process each moment
        for (int i = 1; i <= n; i++) {
            f -= min(a * (m[i] - m[i - 1]), b); // Reduce charge
        }

        // Check if phone still has enough charge
        if (f > 0) {
            cout << "YES" << endl; // Can send all messages
        } else {
            cout << "NO" << endl; // Cannot send all messages
        }
    }

    return 0;
}